### PR TITLE
Added some documentation to some of my classes, renamed StockAPIAccess to StockAPIGateway so our naming conventions are consistent

### PR DESCRIPTION
**Pull request by Spaghetti-12**

### CODE REVIEW

File: src/main/java/BuyStockUseCase/BuyStockGUI.java [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import java.awt.*;
 
 public class BuyStockGUI extends JFrame implements iBuyStockGUI {
+    /**
+     * GUI where the user can purchase stocks
```
Comment (noel-toms) [inline src/main/java/BuyStockUseCase/BuyStockGUI.java:12]: I would recommend adding something simple about the class attributes such as the functionality of the buttons

**Review by Peaceful36 — DISMISSED** [review]
Seems good. I think we should merge this to main quick for everyone to update the changes in the API.

File: src/main/java/BuyStockUseCase/BuyStockPresenter.java [hunk 2]
```diff
@@ -5,6 +5,9 @@
 import java.io.IOException;
 
 public class BuyStockPresenter {
+    /**
+     * Presenter for the buy stock GUI
```
Comment (noel-toms) [inline src/main/java/BuyStockUseCase/BuyStockPresenter.java:9]: Maybe a small comment on what the presenter does. For example what it connects, like GUI to UseCaseInteractor. This is just to show clear architecture is follwed.

File: src/main/java/BuyStockUseCase/PortfolioGUI.java [hunk 3]
```diff
@@ -28,12 +31,15 @@ public class PortfolioGUI extends JFrame implements iPortfolioGUI {
 
     public PortfolioGUI(Portfolio port, String username, boolean isComp) {
         super();
+        /**
+         * GUI where the user can view their portfolio information
+         */
         this.port = port;
         this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```
Comment (noel-toms) [inline src/main/java/BuyStockUseCase/PortfolioGUI.java:38]: Not necessary but I would recommend adding a small comment here between these lines of code just to distinguish what's happening and make it less confusing. Refer to my branch for more clarification, but what I did was split up setting up jlabels and other code with a comment. Please let me know if you would like me to clarify better

**Review by noel-toms — APPROVED** [review]
For now, I will approve this as there is a big change that needs to be merged into the main branch which will impact many Use cases. However, after merging I would recommend you check and read over my comments and let me know about any clarification.

**Comment by noel-toms** [comment]
I recommend waiting for everyone to approve this PR before we merge to ensure everyone is aware of this change. Otherwise, people might be confused why their code isn't working.

**Review by eladperlman — DISMISSED** [review]
Looks good, I was always confused why the naming conventions didn't match.

**Review by eladperlman — DISMISSED** [review]
Looks good!

**Review by noel-toms — DISMISSED** [review]
Approved

**Review by benmarlow — DISMISSED** [review]
Approved. Good work

**Review by eladperlman — APPROVED** [review]
Looks good

**Review by benmarlow — APPROVED** [review]
Approved.

**Review by noel-toms — APPROVED** [review]
Looks Good.

**Pull request merged**